test/odbc/test_pg_datatype.cpp

```diff
@@ -0,0 +1,160 @@
+  };
+
+  vector <string> inserted_values = {
+    "-9223372036854775808",
+    "9223372036854775807",
+    "3",
+    "NULL"
+  };
+
+  string insert_string = ""; 
+  for (int i = 0; i< NUM_ROWS; ++i) {
+    insert_string = insert_string + "(" + std::to_string(i) + "," + inserted_values[i] + ")";
+    if (i+1 != NUM_ROWS)
+      insert_string = insert_string + ", ";
+  }
+
+  // Step 1: Create a table with columns defined with the specific datatype being tested. 
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, BIGINT_TABLE_COLUMNS));
+
+  // Step 2: Insert values into the table using the correct ODBC data type mapping.
+  odbcHandler.ExecQuery(InsertStatement(TABLE_NAME, insert_string));
+  odbcHandler.CloseStmt();
+
+  // Step 3: Select all from the tables and assert that the following attributes of the type is correct:
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {"pk"}));
+
+  // Make sure attributes are correct
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_LENGTH,
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &length); // not the same datatype as documented?
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(length, LENGTH_EXPECTED);
+  
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_PRECISION,
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &precision); 
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(precision, PRECISION_EXPECTED);
+
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_SCALE,
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &scale); 
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(scale, SCALE_EXPECTED);
+
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_TYPE_NAME,
+                          name,
+                          BUFFER_SIZE,
+                          (SQLSMALLINT*) &strlength,
+                          NULL); 
+  ASSERT_EQ(string(name), NAME_EXPECTED);
+
+  // Make sure values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  for (int i = 0; i < NUM_ROWS; ++i) {
+    
+    rcode = SQLFetch(odbcHandler.GetStatementHandle());
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+    ASSERT_EQ(pk_len, BYTES_EXPECTED);
+    ASSERT_EQ(pk, i);
+
+    if (inserted_values[i] != "NULL")
+    {
+      ASSERT_EQ(data_len, BYTES_EXPECTED);
+      ASSERT_EQ(data, strtoll(inserted_values[i].c_str(), NULL, 10));
+    } 
+    else 
+      ASSERT_EQ(data_len, SQL_NULL_DATA);
```

Comment:
Could we also somehow assert we only get one `NULL` row back (or however many we inserted)?

Reply:
This is implicitly asserted as the next statement after it exits this loop, I check if it's the end of the result set.